R/db.R

```diff
@@ -7,11 +7,28 @@
 #'
 db_connect <- function(type, path) {
   x <- dettl_db_args(path, type)
-  con <- do.call(DBI::dbConnect,
+  con <- do.call(dbi_db_connect,
                  c(list(x$driver()), x$args))
   con
 }
 
+#' Create a connection a DBMS.
+#'
+#' Uses \code{\link[DBI]{dbConnect}} to connect to a DBMS. If this uses
+#' \code{\link[RSQLite]{SQLite}} then it will ensure foreign key constraints are
+#' enabled.
+#'
+#' @keywords internal
+dbi_db_connect <- function(drv, ...) {
```

Comment:
This way is fine, but it took me a minute to work out what is going on. I think the way we do it in `orderly` is closer to 

```r
db_connect <- function(type, path) {
  x <- dettl_db_args(path, type)
  con <- do.call(DBI::dbConnect,
                 c(list(x$driver()), x$args))
  sqlite_enable_fk(con)
  con
}

sqlite_enable_fk <- function(con) {
  if (class(drv) == "SQLiteDriver") {
    ## Foreign key constraints aren't enabled by default in SQLite. Ensure
    ## they are enabled each time we connect to the db.
    DBI::dbExecute(con, "PRAGMA foreign_keys = ON")
  }
}
```

which I think is slightly easier to follow

Reply:
So I did it this way because the current `dbi_db_connect` method is used in several of my tests to connect to test db. I could move that connect for testing into a helper function, perhaps that will be clearer.

Reply:
it can always be done later if you don't think it's worth it - it's up to you

Reply:
No that's good, I've updated that now.